v2/kafka-common/src/main/java/com/google/cloud/teleport/v2/kafka/transforms/NonWireAvroDeserializer.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.kafka.transforms;
+
+import io.confluent.kafka.serializers.KafkaAvroDeserializer;
+import java.io.IOException;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.Decoder;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.header.Headers;
+
+public class NonWireAvroDeserializer extends KafkaAvroDeserializer {
```

Comment:
nit: The class name doesn't make sense to me. What's "non-wire" deserializer?

Reply:
This deserializer is for messages in non-wire format. How about renaming it to `ByteAvroDeserializer`?

Reply:
What about BinaryAvroDeserializer?

Reply:
I guess let's decide later during refactoring, when we agree on different format we support and Enum values.